.cbmc-batch/stubs/aws_hash_iter_overrides.c

```diff
@@ -9,15 +9,41 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+/* If the consumer of the iterator doesn't use the .element in the iterator, we can just leave it undef
+ * This is sound, as it gives you a totally nondet value every time you call the iterator, and is the default behaviour
+ * of CBMC. But if it is used, we need a way for the harness to specify valid values for the element, for example if
+ * they are copying values out of the table. They can do this by defining
+ * -DHASH_ITER_ELEMENT_GENERATOR=the_generator_fn, wher the_generator_fn has signature:
+ *   the_generator_fn(struct aws_hash_iter *new_iter, const struct aws_hash_iter* old_iter).
+ *
+ * [new_iter] is a pointer to the iterator that will be returned from this function, and the generator function can
+ * modify it in any way it sees fit.  In particular, it can update the [new_iter->element] field to be valid for the
+ * type of hash-table being proved.  Two obvious generators are:
+ *   (a) one that simply creates a new non-determinsitic value each time its called using malloc
+ *   (b) one that uses a value stored in the underlying map, and copies it into the iterator.
```

Comment:
These two do seem useful enough that you might provide implementations for them in one of the common utility files? Not asking for changes now, but something to consider for the future.

Also this is a super-cool idea 😎

Reply:
That sounds like an excellent idea for a future PR